core/node/events/stream_cache.go

```diff
@@ -214,21 +225,41 @@ func (s *streamCacheImpl) CacheCleanup(ctx context.Context, enabled bool, expira
 func (s *streamCacheImpl) tryLoadStreamRecord(
 	ctx context.Context,
 	streamId StreamId,
-	loadView bool,
-) (SyncStream, StreamView, error) {
-	// Same code is called for GetStream, GetSyncStream and CreateStream.
+) (*streamImpl, error) {
 	// For GetStream the fact that record is not in cache means that there is race to get it during creation:
 	// Blockchain record is already created, but this fact is not reflected yet in local storage.
 	// This may happen if somebody observes record allocation on blockchain and tries to get stream
 	// while local storage is being initialized.
 	record, _, mb, err := s.params.Registry.GetStreamWithGenesis(ctx, streamId)
 	if err != nil {
-		return nil, nil, err
+		// Loop here waiting for record to be created.
+		// This is less optimal than implementing pub/sub, but given that this is rare codepath,
+		// it is not worth over-engineering.
+		ctx, cancel := context.WithTimeout(ctx, time.Second*10)
+		defer cancel()
+		delay := time.Millisecond * 20
+	forLoop:
+		for {
+			select {
+			case <-ctx.Done():
+				return nil, ctx.Err()
+			case <-time.After(delay):
+				entry, _ := s.cache.Load(streamId)
+				if entry != nil {
+					return entry.(*streamImpl), nil
+				}
+				record, _, mb, err = s.params.Registry.GetStreamWithGenesis(ctx, streamId)
```

Comment:
Yes, great point!
There is a protection against this scenario in line 271.
I'm gonna commit this fix since it's good enough for single-replica case.
Lets discuss replicated case (it should be handled consistently with other scenarios).